src/helpers/component-tree.ts

```diff
@@ -1,10 +1,14 @@
 import { ReactTestInstance } from 'react-test-renderer';
 
+export type HostReactTestInstance = ReactTestInstance & { type: string };
```

Comment:
Q: Is that actually working? I remember trying to do something similar in the past, but there were some TS issues that made me abandon such approach. Unfortunately I do not remember these issues, they might also be related to older TS or React/RN types versions.

Reply:
What do you mean by is it working? Elements of type HostReactTestInstance have a type property of type string so I'd say it's working but maybe you're thinking of something else? 

It should work fine with `jest-native` yes, matchers expect a ReactTestInstance and a HostReactTestInstance is still of type ReactTestInstance

Reply:
iirc the issue was that `{ type: string }` could not be properly cast onto `ReactTestInstance` which expected a narrower type of `{ type: 'a' | 'span' | ... }`. Unfortunately I have not yet been able to test drive this PR.

Reply:
Indeed the `type` only accepts string literal union values of HTML tags.
<img width="813" alt="image" src="https://github.com/callstack/react-native-testing-library/assets/6368606/24ea9316-8a95-473b-8630-a636e6b9675e">